src/solver/conv_hip_implicit_gemm_3d_grouped_fwd_xdlops.cpp

```diff
@@ -0,0 +1,484 @@
+                                                              args.output,
+                                                              args.out_strides,
+                                                              args.strides,
+                                                              args.dilation,
+                                                              args.lPadding,
+                                                              args.rPadding,
+                                                              {},
+                                                              {},
+                                                              {});
+        if(conv_ptrs[i]->IsSupportedArgument(argument_ptr.get()))
+            return true;
+    }
+    return false;
+}
+
+namespace {
+
+template <typename DataType>
+void RunCKSolution(const Handle& handle,
+                   const AnyInvokeParams& primitive_parameters,
+                   const CKArgs& args,
+                   const std::string& kernel_id)
+{
+    const auto conv_ptrs = DeviceOpGFwdPtrs<DataType>::GetInstances();
+    int i                = 0;
+    for(; i < conv_ptrs.size(); i++)
+    {
+        if(conv_ptrs[i]->GetTypeString() == kernel_id)
+        {
+            break;
+        }
+    }
+    assert(i != conv_ptrs.size());
+    auto& conv_ptr      = conv_ptrs.at(i);
+    auto& data_ctx      = primitive_parameters.CastTo<conv::DataInvokeParams>();
+    const auto& tensors = data_ctx.tensors;
+
+    auto argument_ptr = conv_ptr->MakeArgumentPointer(
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(tensors.in)),
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(tensors.w)),
+        {},
+        static_cast<void*>(tensors.out),
+        args.input,
+        args.in_strides,
+        args.weight,
+        args.wei_strides,
+        {},
+        {},
+        args.output,
+        args.out_strides,
+        args.strides,
+        args.dilation,
+        args.lPadding,
+        args.rPadding,
+        {},
+        {},
+        {});
+    auto invoker_ptr = conv_ptr->MakeInvokerPointer();
+    invoker_ptr->Run(argument_ptr.get(), {handle.GetStream()});
+}
+
+namespace conv {
+
+InvokerFactory
+MakeCK3DGroupFwdInvokerFactory(const miopen::ProblemDescription& problem,
+                               const PerformanceConfigHipImplicitGemm3DGroupFwdXdlops& config)
+{
+    auto args                  = CKArgs{problem};
+    miopenDataType_t data_type = problem.conv_problem.GetInDataType();
+    auto kernel_id             = config.kernel_id;
+
+    return [args, data_type, kernel_id](const std::vector<Kernel>& kernels) {
+        std::ignore = kernels;
+        return [args, data_type, kernel_id](const Handle& handle,
+                                            const AnyInvokeParams& primitive_parameters) {
+            switch(data_type)
```

Comment:
Move switch out of the Invoker, see https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2305#discussion_r1290090595.

Reply:
Switch can be moved out. Please see 9034f3e in #2305